packages/app/src/messages/components/messages/messages.component.tsx

```diff
@@ -172,50 +184,54 @@ const Messages: FunctionComponent<MessagesProps> = ({
   }, [messageLayoutNotifications])
 
   useEffect(() => {
-    const hideInfoPopupsTimeout = setTimeout(() => {
-      messagePopupNotifications
-        .filter((item) => item.method === NotificationMethod.Popup)
-        .forEach((item) => {
-          removeNotification(item.id)
-        })
+    messagePopupNotifications.length > 0 &&
+      removePopupNotification(messagePopupNotifications)
+    threadPopupNotifications.length > 0 &&
+      removePopupNotification(threadPopupNotifications)
+    // AUTO DISABLED - fix me if you like :)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (
+      state === State.Loaded &&
+      messagePopupNotifications.length > 0 &&
+      states.messageDeleting
+    ) {
+      updateFieldState("messageDeleting", false)
+      updateFieldState("messageDeletingConfirmation", false)
+      messagesState === MessagesState.SearchResult && handleSearchMessage()
+    }
+
+    const hideMessageInfoPopupTimeout = setTimeout(() => {
+      removePopupNotification(messagePopupNotifications)
     }, 5000)
     return () => {
-      clearTimeout(hideInfoPopupsTimeout)
+      clearTimeout(hideMessageInfoPopupTimeout)
     }
     // AUTO DISABLED - fix me if you like :)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [messagePopupNotifications])
 
   useEffect(() => {
-    if (state !== State.Loaded) {
-      return
+    if (
+      state === State.Loaded &&
+      threadPopupNotifications.length > 0 &&
+      states.threadDeleting
+    ) {
+      updateFieldState("threadDeleting", false)
+      updateFieldState("threadDeletingConfirmation", false)
     }
 
-    const handleDeletingStateTimeout = setTimeout(() => {
-      if (states.messageDeleting) {
-        updateFieldState("messageDeleting", false)
-        updateFieldState("messageDeletingConfirmation", false)
-        messagesState === MessagesState.SearchResult && handleSearchMessage()
-      }
-
-      if (states.threadDeleting) {
-        updateFieldState("threadDeleting", false)
-        updateFieldState("threadDeletingConfirmation", false)
-        updateFieldState("threadDeletingInfo", true)
-      }
-    }, 1000)
-
-    const hideInfoPopupsTimeout = setTimeout(() => {
-      updateFieldState("threadDeletingInfo", false)
+    const hideThreadInfoPopupTimeout = setTimeout(() => {
+      removePopupNotification(threadPopupNotifications)
```

Comment:
Not a change request, just a thought after time - maybe it would be better to use e.g. https://www.npmjs.com/package/react-toastify to handle popups/notifications? It could handle some of the work for us and it is easy to use.

(or maybe in the past there was a particular reason to not use this).

Reply:
For now, I used what was prepared before, maybe we can think about it if there are more notifications.